### ISSUE TITLE
[Enhancement]: Search users/customers by first name + last name

### ISSUE THREAD

**Issue by webdados** [woothemes/woocommerce#33701]
### Describe the solution you'd like

In several countries, it's common to refer to people by first name + last name and, currently, it's not possible to search in this way with a WP_User_Query.
It's pretty easy to do it via a direct SQL query, but that doesn't make a lot of sense.

Regarding orders, we already have _billing_address_index and _shipping_address_index as meta in place, which solves the problem.

I recommend that you consider doing the same for the user meta, making it possible to search by first name + last name in 3rd party code.

**Comment by nicomollet** [comment]
The `_billing_address_index` postmeta has some strange values in my shop (We disabled `address`, `postalcode`, `city` from the checkout billing form).

Some are `Firstname Lastname Company Email`
Some are just `Email`

So when we search for `Firstname Lastname`, only some of the orders are found.

I don't understand why this field is not always filled the same way.

**Comment by webdados** [comment]
@nicomollet I don't think that problem is related to this ticket, which is requesting to create a similar index on users.

**Comment by nicomollet** [comment]
You are right, you are talking about the users postmeta.
My problem is with Orders postmeta. 👍

**Comment by webdados** [comment]
No problem on the issue being closed. It took so long that I ended up by creating the index on my plugin, where I needed this to exist.
